# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/CanvasRenderer.cpp
# Commit: cdaa8a4e9e36
# Full Hash: cdaa8a4e9e366ad041b429a8f5bd0b9b259d4722
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-09 09:21:34
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - NON_PREMULT TextureClient iff NON_PREMULT CanvasClient. r=handyman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D78800
# ==============================================================================

diff -r 9ff26bcc580c -r cdaa8a4e9e36 gfx/layers/CanvasRenderer.cpp
--- a/gfx/layers/CanvasRenderer.cpp	Mon Jun 08 20:35:38 2020 +0000
+++ b/gfx/layers/CanvasRenderer.cpp	Mon Jun 08 20:58:03 2020 +0000
@@ -46,7 +46,8 @@
   return mData.GetContext() == aData.GetContext();
 }
 
-std::shared_ptr<BorrowedSourceSurface> CanvasRenderer::BorrowSnapshot() const {
+std::shared_ptr<BorrowedSourceSurface> CanvasRenderer::BorrowSnapshot(
+    const bool requireAlphaPremult) const {
   const auto context = mData.GetContext();
   if (!context) return nullptr;
   const auto& provider = context->GetBufferProvider();
@@ -57,7 +58,7 @@
     ss = provider->BorrowSnapshot();
   }
   if (!ss) {
-    ss = context->GetFrontBufferSnapshot();
+    ss = context->GetFrontBufferSnapshot(requireAlphaPremult);
   }
   if (!ss) return nullptr;
 